firmware/ventilator-controller-stm32/Core/Src/Pufferfish/Driver/I2C/HoneywellABP/Sensor.cpp

```diff
@@ -76,6 +70,12 @@ InitializableState Sensor::measure(float &output) {
     retry_count_ = 0;  // reset retries to 0 for next measurement
     output = sample_.pressure;
     next_action_ = fsm_.update();
+    if (sample_.status != ABPStatus::no_error && sample_.status != ABPStatus::stale_data) {
+      return InitializableState::failed;
+    }
+    if (Util::within(sample_.pressure, pmin, pmax)) {
+      return InitializableState::failed;
+    }
```

Comment:
We should definitely check the pressure against `pmin` and `pmax` in `initialize`.

Currently the `SFM3019` `Sensor` class does not check its measured values against `flow_min` and `flow_max` in the `measure` method, only in the `check_range` method (which is part of the setup routine). I'm not sure whether we want `measure` to include range-checking, but we should be consistent across sensors. If we want to include range-checking in `measure`, then we will need to figure out how to handle such errors. We'll probably need to do some risk analysis (and/or look at what was identified and recorded in Matrix Requirements) to determine whether we do want range-checking in `measure`. For now, let's have the HoneywellABP `Sensor` only do range-checking in `initialize`, and I'll file a Github issue about this.